--- a/seat.c
+++ b/seat.c
@@ -17,7 +17,6 @@
 #include <wayland-server-core.h>
 #include <wlr/backend.h>
 #include <wlr/backend/multi.h>
-#include <wlr/backend/session.h>
 #include <wlr/types/wlr_cursor.h>
 #include <wlr/types/wlr_data_device.h>
 #include <wlr/types/wlr_idle_notify_v1.h>
@@ -259,18 +258,7 @@ handle_keybinding(struct cg_server *server, xkb_keysym_t sym)
 		return true;
 	}
 #endif
-	if (server->allow_vt_switch && sym >= XKB_KEY_XF86Switch_VT_1 && sym <= XKB_KEY_XF86Switch_VT_12) {
-		if (wlr_backend_is_multi(server->backend)) {
-			if (server->session) {
-				unsigned vt = sym - XKB_KEY_XF86Switch_VT_1 + 1;
-				wlr_session_change_vt(server->session, vt);
-			}
-		}
-	} else {
-		return false;
-	}
-	wlr_idle_notifier_v1_notify_activity(server->idle, server->seat->seat);
-	return true;
+	return false;
 }
 
 static void
