--- a/meson.build
+++ b/meson.build
@@ -299,11 +299,19 @@
         )
       )
     elif version.version_compare('<' + required)
-      error(
-        '@0@ must be version >= @1@ (found: @2@)'.format(
-          option_name, required, version,
+      if meson.is_cross_build()
+        warning(
+          '@0@ on build host is version @1@ (target needs >= @2@)'.format(
+            option_name, version, required,
+          )
         )
-      )
+      else
+        error(
+          '@0@ must be version >= @1@ (found: @2@)'.format(
+            option_name, required, version,
+          )
+        )
+      endif
     endif
   endif
 endforeach
